activeviam/activeui-migration
# UI-8069 - Make the CLI usable without access to ActiveViam's artifactory

See UI-8069

This PR makes the `activeui-migration` CLI usable without access to ActiveViam's artifactory.
Authorized access is only needed to install and build the CLI yourself if custom code is added.

In particular, all dependencies are moved to dev dependencies.
The bundle that is built with webpack and allows to run the CLI still holds the dependencies that are required at runtime.

**To test**
- Navigate to the `activeui-migration` repo and checkout this branch
- (Optional) Add some `console.log`s in `bin.ts` that will make you comfident that it is indeed your local code that is run
- Run `yarn` and `yarn build` to build the CLI
- Clear your `npx` cache:
  - Run `npm config get cache` to find the npm cache folder on your file system
  - Navigate to it, and locate the `npx` or `_npx` subfolder
  - Delete it
- Run `npm link` to make `activeui-migration` available globally in your file system through `npx`
- Navigate to any unrelated folder outside of your `activeui-migration` repo, e.g. `cd ~`
- Run `npx -- activeui-migration migrate --help`
=> You should see information on the CLI usage, as well as the logs that you added in the optional step, if any.
